### PR TITLE
Questions and answers notifications

### PR DESCRIPTION
**Pull request by abdelrahman-elkady**
- Notify user on up/down votes
- Notify Question creator when he receives new answers
- Notify Question/Answer creator about new comments on his entry
- Notify commenters on some entry about new comments
- Notify Answer owner if his answer is selected as best answer

### CODE REVIEW

File: client/templates/discussions/comment/commentEditModal.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Template.commentEditModal.helpers({
+  commentData() {
```
Comment (alihassan0) [inline client/templates/discussions/comment/commentEditModal.js:2]: we should agree on a convention for naming such helpers

Comment (abdelrahman-elkady) [inline client/templates/discussions/comment/commentEditModal.js:2]: only this ?

All templates :smiley:

File: lib/methods/comments.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+      content: content,
+      createdAt: Date.now()
+    });
+
+    let userId = Meteor.userId();
+
+    if (!commentId)
+      throw new Meteor.Error(500, "Something went wrong, please try again later");
+
+    // FIXME: High priority refactoring task
```
Comment (alihassan0) [inline lib/methods/comments.js:16]: i suggest using templates for the html markup

Comment (abdelrahman-elkady) [inline lib/methods/comments.js:16]: all of these notification insertion will not be there in the next release inshaAllah

and yes, we can't keep that into js files anymore

**Pull request merged**